### ISSUE TITLE
Split Exposed in multi sub module

### ISSUE THREAD

**Issue by chmuche** [JetBrains/Exposed#184]
I think exposed is a great Orm lite / query builder
But It's possible to split Exposed in multi jar?

For example

```
/exposed-sql-core
This module offer only the Table et Column API without Date,Time and DataTime columnType
```

```
/exposed-sql-compat --> depends [exposed-sql, joda-time]
This module create Column API with Date,Time and DataTime columnType using Joda Time 
```

```
/exposed-sql --> depends [exposed-sql]
This module create Column API with Date,Time and DataTime columnType using java.time API
```

```
/exposed-dao-core --> depends [exposed-sql]
This module offer the Dao API without Date, Time and DateTime property
```

```
/exposed-dao-compat --> depends [exposed-dao-core, joda-time]
This module create Column API with Date,Time and DataTime property using Joda Time 
```

```
/exposed-dao --> depends [exposed-dao-core]
This module create Column API with Date,Time and DataTime property using java.time API 
```

**Comment by lnhrdt** [comment]
This might be a good solution to supporting the java.time API in Java 1.8 while maintaining Java 1.6 support (more discussion in #100).

@Tapac what's your perspective? You shared a different idea in #100 that would resolve class availability at runtime.

**Comment by Tapac** [comment]
@lnhrdt , I'll think about. 
Possibly it (split into modules) is more "right" way to support different date/time providers, thank you that you notice.

**Comment by Tapac** [comment]
Released in 0.18.1
https://github.com/JetBrains/Exposed/wiki/LibDocumentation#exposed-0181-and-higher
